### ISSUE TITLE
Make Editor Viewport Message into a Label Node

### ISSUE THREAD

**Issue by ryevdokimov** [godotengine/godot-proposals#8789]
### Describe the project you are working on

N/A

### Describe the problem or limitation you are having in your project

![image](https://github.com/godotengine/godot-proposals/assets/105675984/692cec28-4d0c-4b26-8822-7555a85198ec)

Currently this text is rendered directly into the viewport, and there is no way to access it inside the editor. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think this would align more with Godot's philosophy of "The Godot Editor is a Godot Game":

https://docs.godotengine.org/en/stable/getting_started/introduction/godot_design_philosophy.html#the-godot-editor-is-a-godot-game

I could see situations where the text in that field could be modified or exposed. I personally could use it as a stopgap for: #8445, by using the text that will be exposed to indirectly know that the gizmo is being manipulated. 

It would also allow users to easily move, delete or add their own messages if they'd like. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There are already label nodes on the viewports: 

![image](https://github.com/godotengine/godot-proposals/assets/105675984/504f42f1-7ece-4be0-8878-681bdd627314)

It could be reused or added another similar node.

Inside node_3d_editor_plugin.cpp for example.

![image](https://github.com/godotengine/godot-proposals/assets/105675984/2a24b295-8a6e-44fb-a380-d2f8b500cbbb)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

N/A

**Comment by Calinou** [comment]
Exposing this text should be done through EditorInterface, as opposed to encouraging the use of editor nodes through `get_node()` (which has zero compatibility guarantees).

There may be good reasons to move that text's drawing to a node, but I don't think plugins are one of them.

**Comment by ryevdokimov** [comment]
> Exposing this text should be done through EditorInterface, as opposed to encouraging the use of editor nodes through `get_node()` (which has zero compatibility guarantees).
> 
> There may be good reasons to move that text's drawing to a node, but I don't think plugins are one of them.

Fair point on compatibility, but I suppose that's true of all nodes that the editor is comprised of, it's still been useful to have them all exposed despite the risk. 

I suppose I could have started a discussion on why some things are messages, and others label nodes (like when overriding a material). 

As I see it now, you can't override a material and do anything else that shows up as a message at the same time, so all of that could have been consolidated. 

Why not choose one or the other to simplify?

**Comment by KoBeWi** [comment]
> I think this would align more with Godot's philosophy of "The Godot Editor is a Godot Game":

Well, the text is drawn using `draw_string()`, which is also available for games. It just doesn't use Label node, because there is no reason to do so. And it doesn't need to be a Label to get exposed.

**Comment by ryevdokimov** [comment]
> > I think this would align more with Godot's philosophy of "The Godot Editor is a Godot Game":
> 
> Well, the text is drawn using `draw_string()`, which is also available for games. It just doesn't use Label node, because there is no reason to do so. And it doesn't need to be a Label to get exposed.

Understood on the first part, however my understanding is you can't use editor plugins to make it **not** do that. If it were a label, you make it do anything. Put it in a different corner, anywhere or nowhere. Assuming it's not exposed through an EditorInterface, the values could be used despite compatibility risks. The values could be converted and displayed as fractions when using imperial units.

I've been personally modifying the editor itself for use as an engineering tool, and I can see others doing the same for other scientific/engineering tool reasons without the requirement of compiling something into the game.  

This would take just reusing the material override label node that is already there and remove the draw calls unless I'm missing something.

**Comment by KoBeWi** [comment]
> Understood on the first part, however my understanding is you can't use editor plugins to make it not do that.

Well, yeah, if it's a Label, you can find the node and make it do anything. You can hack many editor elements like that. However such hacks are unstable, there is no guarantee that it will work at any time and can break the editor if used carelessly. Hence suggesting a change to enable hacks is not something that can be considered.

If you only want to modify the text (and hiding is the same as setting empty text), it can be exposed as a method that overrides the drawn string (like `set_viewport_hint_text(text)`). Though if you wanted to reposition the text or change its style, exposing it as Label would make more sense.

**Comment by ryevdokimov** [comment]
> > Understood on the first part, however my understanding is you can't use editor plugins to make it not do that.
> 
> Well, yeah, if it's a Label, you can find the node and make it do anything. You can hack many editor elements like that. However such hacks are unstable, there is no guarantee that it will work at any time and can break the editor if used carelessly. Hence suggesting a change to enable hacks is not something that can be considered.
> 
> If you only want to modify the text (and hiding is the same as setting empty text), it can be exposed as a method that overrides the drawn string (like `set_viewport_hint_text(text)`). Though if you wanted to reposition the text or change its style, exposing it as Label would make more sense.

Yes, foot gun is always a risk when exposing things to users. However, I'd argue that you're already in the weeds if you're messing with Editor Plugins. I could crash my whole computer writing scripts in Godot if I wanted to. 

My proposal isn't about necessary hacking things together, although I do personally encourage it if you're limited to the confines of the engine. It's to suggest that such a change would be consistent with my interpretation of Godot's design philosophy. The editor is a game itself, let users have as much control on how it looks without having to modify the engine code directly. Don't forcibly draw things unless it is absolutely fundamental for functionality. Especially when the editor already has similar elements that already do this.  (Not saying I'm correct here, just discussing).
